include/ares_dns_record.h

```diff
@@ -961,6 +961,18 @@ CARES_EXTERN ares_status_t ares_dns_parse(const unsigned char *buf,
  */
 CARES_EXTERN ares_status_t ares_dns_write(ares_dns_record_t *dnsrec,
                                           unsigned char **buf, size_t *buf_len);
+
+/*! Search for a complete DNS message
+ *
+ *  \param[in] channel  Pointer to channel on which queries will be sent.
+ *  \param[in] dnsrec   Pointer to initialized and filled DNS record object.
+ *  \param[in] callback Callback function invoked on completion or failure of
+ *                      the query sequence.
+ *  \param[in] arg      Additional argument passed to the callback function.
+ */
+CARES_EXTERN void ares_search_dnsrec(ares_channel_t *channel,
+                                     ares_dns_record_t *dnsrec,
+                                     ares_callback callback, void *arg);
```

Comment:
I have done this.

Note that the function uses `ares_dns_record_t` which is defined in ares_dns_record.h. I have had to re-order the contents of ares.h so that ares_dns_record.h is included beforehand. Also, ares_dns_record.h uses `struct ares_addr`, so I needed to move this up too.

An alternative could be to move 
```
/*! Opaque data type representing a DNS Packet */
struct ares_dns_record;

/*! Typedef for opaque data type representing a DNS Packet */
typedef struct ares_dns_record ares_dns_record_t;
```
from ares_dns_record.h to ares.h.